src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/functions.ts

```diff
@@ -333,16 +335,12 @@ export function checkFunctionInvocationComplete(
   }
 
   // If the function is complete, check that the types of the arguments match the function definition
-  const hasCorrectTypes = fnDefinition.signatures.some((def) => {
+  const hasCorrectTypes = fnDefinition.signatures.some(({ params }) => {
     return func.args.every((a, index) => {
-      return (
-        fnDefinition.name.endsWith('null') ||
-        def.params[index].type === 'any' ||
-        def.params[index].type === getExpressionType(a) ||
-        // this is a special case for expressions with named parameters
-        // e.g. "WHERE field == ?value"
-        isParamExpressionType(getExpressionType(a))
-      );
+      const expectedType = params[index].type;
+      const actualType = getExpressionType(a);
+
+      return argMatchesParamType(actualType, expectedType, isLiteral(a), false);
```

Comment:
Are you sure we shouldn't accept `unknown` values here? Usually when we're uncertain about a type, we allow it to match signatures.

Reply:
Here `acceptUnknown` is being set to `false`, so `argMatchesParamType` is configured not to accept `unknown`-type arguments.